### ISSUE TITLE
yolo12

### ISSUE THREAD

**Issue by cainiao123s** [ultralytics/ultralytics#18759]
### Search before asking

- [x] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

Hello, I would like to ask how much performance improvement is needed to become YOLO12? I have currently conducted experiments on the CIFAR100 dataset, and the performance of YOLOv8 is better than that of YOLOv11. My method has improved by 0.38% compared to YOLOv8n.

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cainiao123s, thank you for your interest in Ultralytics 🚀! We appreciate you taking the time to share your insights and experiments. YOLO's evolution, including hypothetical versions like "YOLO12," involves ongoing development and research to improve performance across diverse datasets and tasks.

To ensure that your developments and proposed improvements integrate seamlessly with YOLO, we recommend reviewing our [contributing guidelines](https://github.com/ultralytics/ultralytics/blob/main/CONTRIBUTING.md) and exploring resources in the [Docs](https://docs.ultralytics.com). You might find these pages particularly useful for customizations and performance enhancements: 
- [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) 
- [Python Usage](https://docs.ultralytics.com/usage/python/)

If you're running into issues or bugs in your development, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can effectively assist you.

Additionally, feel free to connect with the Ultralytics community for feedback and discussions:
- Real-time help on [Discord](https://discord.com/invite/ultralytics) 🎧
- Community insights on [Discourse](https://community.ultralytics.com)
- In-depth threads on our [Subreddit](https://reddit.com/r/Ultralytics)

### Upgrade

Before proceeding further, make sure you’re using the latest version of the `ultralytics` package. You can upgrade with the following command:

```bash
pip install -U ultralytics
```

This ensures you are testing against the latest updates and improvements.

### Environments

YOLO can be leveraged in various environments optimized for machine learning, including:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

### Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

The CI tests verify the correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across macOS, Windows, and Ubuntu. Seeing this green ensures YOLO meets quality standards and should work seamlessly in tested configurations.

This is an automated response to help get you started. An Ultralytics engineer will review your query and provide further assistance soon 🚀😊!

**Comment by glenn-jocher** [comment]
@cainiao123s thank you for your interest and contribution! At this time, there are no plans or timelines for YOLO12, as YOLO11 is the latest release. However, advancements and community-driven improvements, like yours, are always appreciated. Please feel free to submit a PR with your method; we'd be happy to review it and discuss its potential integration.

**Comment by cainiao123s** [comment]
In the image classification task, YOLO did not generate a PR curve, here are my results:

**Comment by glenn-jocher** [comment]
@cainiao123s thank you for sharing your results and progress! It's great to see your advancements. For further experimentation, testing on the COCO dataset is an excellent idea, as it provides a robust benchmark for evaluating model performance. Regarding communication, feel free to continue sharing updates and insights here on GitHub or submit a PR if you'd like to contribute your improvements for review. We look forward to seeing more of your work!

**Comment by cainiao123s** [comment]
Due to computational limitations, I need to adjust the hyperparameter "imgsz." What do you think I should adjust it to in order to save computational power as much as possible while keeping the experiment effective?

**Comment by glenn-jocher** [comment]
To save computational power while maintaining effectiveness, reduce `imgsz` to the smallest resolution that still captures sufficient detail for your dataset, such as 320 or 416. Test the model's performance to ensure it meets your requirements at the adjusted size.

**Comment by cainiao123s** [comment]
> 为了在保持有效性的同时节省计算能力，请将分辨率降低`imgsz`到仍能捕获数据集足够细节的最小分辨率，例如 320 或 416。测试模型的性能以确保它在调整后的尺寸下满足您的要求。
Hi, to become YOLO12, should I consider the computational speed and complexity of improvements while enhancing accuracy? I have proposed a new module, which has achieved good results on the COCO dataset (yolo11 and yolo8 + new module), a lightweight model that is simple yet very effective. Will such a lightweight model be accepted as YOLO12? I have not yet published a paper; if I publish a paper (YOLO12), should I submit the PR first for review and then publish the paper, or publish the paper first and then submit the PR? Or should both be done simultaneously?

**Comment by glenn-jocher** [comment]
Thank you for your question and contribution! While YOLO12 is not currently planned, community contributions are always welcome. You can proceed in any order—submit a PR first, publish your paper, or do both simultaneously. If submitting a PR, ensure your changes align with the repository's contribution guidelines. We look forward to seeing your work!

**Comment by cainiao123s** [comment]
When conducting experiments on the cooc dataset, I found that the performance of yolo8 is always significantly better than that of yolo11, and this phenomenon is also reflected in the cifar100 dataset.

If there are no plans to release YOLO12, could it be considered to be accepted as YOLO11X?

**Comment by glenn-jocher** [comment]
Thank you for sharing your findings! While there are no immediate plans for YOLO12 or additional YOLO11 variants like YOLO11X, you're welcome to submit a detailed PR with your improvements for review. This allows the community to evaluate your work and consider its integration into the repository.
